src/core/core.datasetController.js

```diff
@@ -1112,23 +1053,20 @@ export default class DatasetController {
 	_onDataPush() {
```

Comment:
I think the fact that we tried to do something more performant with these methods is undocumented. Are we still doing something special here? If so, it might be worth documenting. If not, then we might not need these methods anymore.

Reply:
Looks like it was done to property animate the new data. I don't think its needed any more.

Reply:
Do you want to go ahead and remove that as part of this PR?

Reply:
I think the point of these methods was to figure out when the array changed. We define an override of `Array.prototype.push` for example that calls `onDataPush`.

Thinking about it more, perhaps we need some kind of option for this? If a user is working in an immutable framework like react, then the existing code worked without issue. It only didn't work when these objects were mutated. If we had a way to go back to the old behaviour or even a way to force an update & reparse we could keep the old code for performance.

Reply:
I actually think we can't remove the data-array overrides, because then new elements would always be added to the end of array and animations would not reflect the real change (shift, unshift or splice).